src/main/java/org/thoughtcrime/securesms/WebxdcStoreActivity.java

```diff
@@ -0,0 +1,135 @@
+package org.thoughtcrime.securesms;
+
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Bundle;
+import android.view.MenuItem;
+import android.webkit.WebResourceRequest;
+import android.webkit.WebResourceResponse;
+import android.webkit.WebSettings;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+
+import androidx.appcompat.app.ActionBar;
+
+import com.b44t.messenger.DcContext;
+import com.b44t.messenger.rpc.HttpResponse;
+import com.b44t.messenger.rpc.Rpc;
+
+import org.thoughtcrime.securesms.connect.DcHelper;
+import org.thoughtcrime.securesms.providers.PersistentBlobProvider;
+import org.thoughtcrime.securesms.util.MediaUtil;
+import org.thoughtcrime.securesms.util.Prefs;
+
+import java.io.ByteArrayInputStream;
+import java.util.HashMap;
+
+public class WebxdcStoreActivity extends PassphraseRequiredActionBarActivity {
+  private static final String TAG = WebxdcStoreActivity.class.getSimpleName();
+
+  private DcContext dcContext;
+  private Rpc rpc;
+
+  @Override
+  protected void onCreate(Bundle state, boolean ready) {
+    setContentView(R.layout.web_view_activity);
+    rpc = DcHelper.getRpc(this);
+    dcContext = DcHelper.getContext(this);
+    WebView webView = findViewById(R.id.webview);
+
+    ActionBar actionBar = getSupportActionBar();
+    if (actionBar != null) {
+      actionBar.setDisplayHomeAsUpEnabled(true);
+      actionBar.setTitle(R.string.webxdc_apps);
+    }
+
+    webView.setWebViewClient(new WebViewClient() {
+      @Override
+      public boolean shouldOverrideUrlLoading(WebView view, String url) {
+        view.loadUrl(url);
+        return true;
+      }
+
+      @TargetApi(Build.VERSION_CODES.N)
+      @Override
+      public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request) {
+        view.loadUrl(request.getUrl().toString());
+        return true;
+      }
```

Comment:
Would be nice to `openUrlInBrowser()` links that don't end on `.xdc`

Reply:
I like the idea, but also the page shouldn't have all that big banners etc

Reply:
I just implemented it, now random URLs are not open in the webview but externally, only clicks in xdc files are intercepted, anyways the website should remove all that links but it is "good to have" for potential 3rd party URLs since the URL can be changed

Reply:
well, staying inside the webview might also be useful for a static-html-store using several .html pages "alternative" - that could just be the file view from apache or nginx then. 

but yeah, just now it is better to open externally, these "alternatives" do not exist, but webxdc.org/apps with source-urls does